pkg/kubelet/dockertools/labels.go

```diff
@@ -190,23 +207,90 @@ func getInt64PointerFromLabel(labels map[string]string, label string) (*int64, e
+// The old labels are added before v1.2, and should be completely removed after we drop support for v1.2.
+// TODO(random-liu): Remove this function when we drop support for v1.2.
+func supplyContainerInfoWithV12Label(labels map[string]string, containerInfo *labelledContainerInfo) {
+	var err error
+	// Only try to get information when there are no corresponding new labels.
+	if _, found := labels[podDeletionGracePeriodLabel]; !found {
+		containerInfo.PodDeletionGracePeriod, err = getInt64PointerFromLabel(labels, oldPodDeletionGracePeriodLabel)
+		if err != nil {
+			logLabelError(*containerInfo, oldPodDeletionGracePeriodLabel, err)
+		}
+	}
+	if _, found := labels[podTerminationGracePeriodLabel]; !found {
+		containerInfo.PodTerminationGracePeriod, err = getInt64PointerFromLabel(labels, oldPodTerminationGracePeriodLabel)
+		if err != nil {
+			logLabelError(*containerInfo, oldPodTerminationGracePeriodLabel, err)
+		}
+	}
+	if _, found := labels[containerRestartCountLabel]; !found {
+		containerInfo.RestartCount, err = getIntegerFromLabel(labels, oldContainerRestartCountLabel)
+		if err != nil {
+			logLabelError(*containerInfo, oldPodTerminationGracePeriodLabel, err)
+		}
+	}
+	if _, found := labels[containerTerminationMessagePathLabel]; !found {
+		containerInfo.TerminationMessagePath = getStringFromLabel(labels, oldContainerTerminationMessagePathLabel)
+	}
+	if _, found := labels[containerPreStopHandlerLabel]; !found {
+		preStopHandler := &api.Handler{}
+		found, err = getJsonObjectFromLabel(labels, oldContainerPreStopHandlerLabel, preStopHandler)
+		if err != nil {
+			logLabelError(*containerInfo, oldContainerPreStopHandlerLabel, err)
+		} else if found {
+			containerInfo.PreStopHandler = preStopHandler
+		}
+	}
+}
+
+// Backward compatible labels for kubernetes v1.1 and before.
+// TODO(random-liu): Keep this for old containers, remove this when we drop support for v1.1.
+const (
+	podLabel = "io.kubernetes.pod.data"
+)
+
+// The label podLabel is added a long time ago (#7421), it serialized the whole api.Pod to a docker label.
 // We want to remove this label because it serialized too much useless information. However kubelet may still work
 // with old containers which only have this label for a long time until we completely deprecate the old label.
 // Before that to ensure correctness we have to supply information with the old labels when newly added labels
 // are not available.
-// TODO(random-liu): Remove this function when we can completely remove label kubernetesPodLabel, probably after
+// TODO(random-liu): Remove this function when we can completely remove label podLabel, probably after
 // dropping support for v1.1.
-func supplyContainerInfoWithOldLabel(labels map[string]string, containerInfo *labelledContainerInfo) {
+func supplyContainerInfoWithV11Label(labels map[string]string, containerInfo *labelledContainerInfo) {
```

Comment:
Is it possible to collapse this method with `supplyContainerInfoWithV12Label()`?